flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResource.java

```diff
@@ -93,6 +93,10 @@ private MiniClusterResource(
 		this.enableClusterClient = enableClusterClient;
 	}
 
+	public MiniClusterType getMiniClusterType() {
```

Comment:
Why not renaming it to sth like `isLegacyDeployment()` and return a `boolean` this will also allow to make the `MiniClusterType` a private class which is also better pattern.

Reply:
I think enums are preferably over booleans.

Reply:
I would prefer to keep the enum. Name `isLegacyDeployment` would deprecate faster then the enum type. Also enums are more flexible (adding/removing more values in the future).

Regardless there is no big advantage of one over the other, so if you want, I can change it either way.

Reply:
This is definitely not a big issue. I will merge.